pliers/stimuli/api.py

```diff
@@ -0,0 +1,78 @@
+import os
+
+from .base import load_stims
+from .compound import CompoundStim
+from .image import ImageStim
+from .text import TextStim
+from .video import VideoStim
+from pliers.transformers import EnvironmentKeyMixin
+
+
+class TweetStimFactory(EnvironmentKeyMixin):
+
+    '''
+    An object from which to generate TweetStims, creates an Api instance from
+    the python-twitter library
+
+    Args:
+        consumer_key (str): A valid consumer key for the Twitter API
+        consumer_secret (str): A valid consumer secret key for the Twitter API
+        access_token_key (str): A valid access token for the Twitter API
+        access_token_secret (str): A valid access token secret for the
+            Twitter API
+
+    To get these credentials, visit https://dev.twitter.com/.
+    '''
+
+    _env_keys = ('TWITTER_CONSUMER_KEY', 'TWITTER_CONSUMER_SECRET',
+                 'TWITTER_ACCESS_TOKEN_KEY', 'TWITTER_ACCESS_TOKEN_SECRET')
+
+    def __init__(self, consumer_key=None, consumer_secret=None,
+                 access_token_key=None, access_token_secret=None):
+        import twitter
```

Comment:
We may want to use the pattern of importing within a try/except block at the top and setting to `None` if import isn't found, and then checking the `twitter` variable here and raising a comprehensible exception for the user if not available.